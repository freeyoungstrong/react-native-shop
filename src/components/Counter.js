import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, View, Text, TouchableOpacity, ImageBackground } from 'react-native';

import { increaseCouner, decreaseCouner } from '../actions/counterActions';

class Counter extends Component {
    pic = {
        uri:
            'https://images.unsplash.com/photo-1519455953755-af066f52f1a6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=634&q=80',
    };

    render() {
        return (
            <ImageBackground source={this.pic} style={styles.image}>
                <View style={styles.container}>
                    <TouchableOpacity onPress={() => this.props.increaseCouner()}>
                        <Text style={styles.text}>Increase</Text>
                    </TouchableOpacity>
                    <Text style={styles.text}>{this.props.counter}</Text>
                    <TouchableOpacity onPress={() => this.props.decreaseCouner()}>
                        <Text style={styles.text}>Decrease</Text>
                    </TouchableOpacity>
                </View>
            </ImageBackground>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
    },
    text: {
        fontSize: 50,
        fontWeight: 'bold',
        textAlignVertical: 'center',
        color: 'white',
    },
    image: {
        width: '100%',
        height: '100%',
        justifyContent: 'center',
    },
});

function mapStateToProps(state) {
    return {
        counter: state.counter,
    };
}

function mapDispatchToProps(dispatch) {
    return {
        increaseCouner: () => dispatch(increaseCouner()),
        decreaseCouner: () => dispatch(decreaseCouner()),
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Counter);
