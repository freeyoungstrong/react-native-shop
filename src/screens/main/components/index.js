export * from './CategoriesList/CategoriesList';
export * from './ProductsList/ProductsList';
