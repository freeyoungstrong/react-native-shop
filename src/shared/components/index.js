export * from './Button/button';
export * from './Input/input';
export * from './NavQuestion/navQuestion';
export * from './Modal/modal';
export * from './Divider/divider';
