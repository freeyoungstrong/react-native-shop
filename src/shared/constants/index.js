export const routes = {
    LOGIN: 'Login',
    REGISTRATION: 'Registration',
    MAIN: 'Main',
    PRODUCT_DETAILS: 'ProductDetails',
    WELCOME: 'Welcome',
    HOME: 'Home',
    MAP: 'Map',
    CART: 'Cart',
    USER_PROFILE: 'UserProfile',
};

export const USER = 'USER';

export const API_URL = 'http://34.73.95.65/index.php';
