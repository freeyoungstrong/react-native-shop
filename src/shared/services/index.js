export * from './products';
export * from './addToCart';
export * from './user';
