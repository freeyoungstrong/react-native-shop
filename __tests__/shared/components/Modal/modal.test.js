import React from 'react';
import { shallow } from 'enzyme';

import { Modal } from 'shared/components';

const props = { visible: false, onClose: jest.fn(), title: 'Title', description: 'Description' };

describe('ModalView component', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Modal {...props} />);
    });

    it('should render fine', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should close Modal', () => {
        wrapper.find('Button').simulate('press');
        expect(props.onClose).toHaveBeenCalled();
    });
});
