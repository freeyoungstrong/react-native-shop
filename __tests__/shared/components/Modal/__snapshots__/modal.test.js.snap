// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalView component should render fine 1`] = `
<Component
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 32,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 5,
          "height": 200,
          "justifyContent": "space-evenly",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Roboto-Medium",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Title
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        Description
      </Text>
      <Button
        onPress={[Function]}
        title="OK"
      />
    </View>
  </View>
</Component>
`;
